packages/xo-web/src/common/smart-backup-pattern.js

```diff
@@ -1,35 +1,59 @@
 import * as CM from 'complex-matcher'
-import { flatten, identity, map } from 'lodash'
+import { flatten, get, identity, isEmpty, map } from 'lodash'
 
 import { EMPTY_OBJECT } from './utils'
 
 export const destructPattern = (pattern, valueTransform = identity) =>
   pattern && {
-    not: !!pattern.__not,
-    values: valueTransform((pattern.__not || pattern).__or),
+    values: valueTransform(
+      pattern.__and !== undefined ? pattern.__and[0].__or : pattern.__or
+    ),
+    notValues: valueTransform(
+      pattern.__and !== undefined
+        ? pattern.__and[1].__not.__or
+        : get(pattern, '__not.__or')
+    ),
   }
 
 export const constructPattern = (
-  { not, values } = EMPTY_OBJECT,
+  { values, notValues } = EMPTY_OBJECT,
   valueTransform = identity
 ) => {
-  if (values == null || !values.length) {
+  const valuesExist = !isEmpty(values)
+  const notValuesExist = !isEmpty(notValues)
+
+  if (!valuesExist && !notValuesExist) {
     return
   }
 
-  const pattern = { __or: valueTransform(values) }
-  return not ? { __not: pattern } : pattern
+  const valuesPattern = valuesExist && { __or: valueTransform(values) }
+  const notValuesPattern = notValuesExist && {
+    __not: { __or: valueTransform(notValues) },
+  }
+  return valuesPattern && notValuesPattern
+    ? { __and: [valuesPattern, notValuesPattern] }
+    : valuesPattern || notValuesPattern
```

Comment:
This could return the value `false`, which is an issue, right?

Reply:
It's handled in line 25